[EFA] Upgrade EFA from 1.17.2 to 1.18.0

EddyMM opened the pull request:
### Description of changes
Upgrades EFA installer from 1.17.2 to 1.18.0
- Efa-driver: `efa-1.16.0-1`
- Efa-config: `efa-config-1.11-1`
- Efa-profile: `efa-profile-1.5-1`
- Libfabric-aws: `libfabric-aws-1.16.0~amzn4.0-1`
- Rdma-core: `rdma-core-41.0-2`
- Open MPI: `openmpi40-aws-4.1.4-2`

### Tests
* Created a Custom AMI with the cookbook changes and confirm EFA (`1.18.0`) was successfully installed

### References
* [EFA installer](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-verify.html)

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

chenwany commented on CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ This file is used to list changes made in each version of the AWS ParallelCluste
 - Upgrade third-party cookbook dependencies:
   - selinux-6.0.5 (from selinux-6.0.4)
   - nfs-5.0.0 (from nfs-2.6.4)
+- Upgrade EFA to `1.18.0`
```
Can we change it to:
Upgrade EFA installer to version 1.18.0
and also mentioned the version of   - EFA driver, EFA configuration, EFA profile, Libfabric, RDMA core, Open MPI version? See an example from this PR: https://github.com/aws/aws-parallelcluster-cookbook/pull/1481/files

EddyMM committed:
[EFA] Upgrade EFA installer to 1.18.0

- Efa-driver: `efa-1.16.0-1`
- Efa-config: `efa-config-1.11-1`
- Efa-profile: `efa-profile-1.5-1`
- Libfabric-aws: `libfabric-aws-1.16.0~amzn4.0-1`
- Rdma-core: `rdma-core-41.0-2`
- Open MPI: `openmpi40-aws-4.1.4-2`

Signed-off-by: Eddy Mwiti <eddmwiti@amazon.com>
